### PR TITLE
Throttle ADR

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
# Description

Removes `throttle.md` in favor of a more complete ADR for throttling

## Linked issues

Closes: https://github.com/cosmos/interchain-security/issues/615
See https://github.com/cosmos/ibc/pull/924 for changes to spec needed for throttling

## Type of change

- [x] `Docs`: Adds documentation

### CODE REVIEW

File: docs/architecture/adr-002-throttle.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+First, we define the following:
+
+* A consumer initiated slash attack "starts" when the first slash packet from such an attack is received by the provider.
+* The "initial validator set" for the attack is the set that existed when the attack started.
+* There is a list of honest validators s.t if they are jailed, `X`% of the initial validator set will be jailed.
+* The "final slashed validator" is the last element in the list of honest validators.
+
+For the following invariant to hold, these assumptions must be true:
+
+* We assume the total voting power of the chain (as a function of delegations) does not increase over the course of the attack.
```
Comment (smarshall-spitzbart) [inline docs/architecture/adr-002-throttle.md:109]: TODO: look into Ivan's derivation

File: docs/architecture/adr-002-throttle.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+# ADR 001: Key Assignment
```
Comment (smarshall-spitzbart) [inline docs/architecture/adr-002-throttle.md:1]: TODO: update doc with semantics changes from #687

File: docs/architecture/adr-002-throttle.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+This invariant is useful because it allows us to reason about the time it takes to jail a certain percentage of the initial provider validator set from consumer initiated slash requests. For example, if `SlashMeterReplenishFraction` is set to 0.06, then it takes no less than 4 replenishment periods to jail 33% of the initial provider validator set on the Cosmos Hub. Note that as of writing this on 11/29/22, the Cosmos Hub does not have a validator with more than 6% of total voting power.
+
+Note also that 4 replenishment period is a worst case scenario that depends on well crafted attack timings.
```
Comment (smarshall-spitzbart) [inline docs/architecture/adr-002-throttle.md:124]: TODO: address the fact that unjailing or not does not break the invariant

**Comment by smarshall-spitzbart** [comment]
TODO: link to Q and A (on wiki section) with attack vectors like #685

**Comment by smarshall-spitzbart** [comment]
@ivan-gavran I've incorporated my understanding of your invariant derivation into this PR, hopefully the `intuition` section is now more understandable

File: docs/architecture/adr-002-throttle.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
+# ADR 001: Key Assignment
```
Comment (smarshall-spitzbart) [inline docs/architecture/adr-002-throttle.md:1]: Title

File: docs/architecture/adr-002-throttle.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+* 2023-01-26: Initial Draft
+
+## Status
+
+Accepted
+
+## Context
+
+The CCV spec is based around the assumption that the provider binary and all consumers binaries are non-malicious, and follow the defined protocols. In practice, this assumption may not hold. A malicious consumer binary could potentially include code which is able to send many downtime or double signing packets at once to the provider.
```
Comment (smarshall-spitzbart) [inline docs/architecture/adr-002-throttle.md:13]:
```suggestion
The CCV spec is based around the assumption that the provider binary and all consumers binaries are non-malicious, and follow the defined protocols. In practice, this assumption may not hold. A malicious consumer binary could potentially include code which is able to send many slash/jail packets at once to the provider.
```

**Comment by smarshall-spitzbart** [comment]
> > @ivan-gavran I've incorporated my understanding of your invariant derivation into this PR, hopefully the `intuition` section is now more understandable
> 
> Cool! It is indeed much clearer now. I noticed a few things that could still be improved and created a [PR](https://github.com/cosmos/interchain-security/pull/714) since I don't have access to push directly. Ping me if you want to discuss the changes.
> 
> One question occurred to me: why not using [Latex rendering](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions) that Github supports now for ADRs?
> 
> Finally, a bit of nitpicking: why calling the property you are describing an invariant? An invariant would be something that holds always (e.g., a simple one, _it is always the case that the meter's value is smaller than its allowance_), whereas what you are describing here is a property.

Thanks for the feedback! On latex rendering, I wasn't aware of this, I'll take a look 👍 

On property vs invariant, I agree with your suggestion and have changed instances of "invariant" to "property"

**Review by MSalopek — APPROVED** [review]

**Review by jtremback — APPROVED** [review]
LGTM

**Pull request merged**